components/modals/ProfilePreview.js

```diff
@@ -1,18 +1,42 @@
 import UserPage from "@components/user/UserPage";
+import { useEffect, useRef } from "react";
 
 export default function Preview({ toggle, data }) {
   const BASE_URL = process.env.NEXT_PUBLIC_BASE_URL;
+  const modalContentRef = useRef(null)
+
+
+  const handleClickOutsideContentBox = (event) => {
+    if (modalContentRef.current && !modalContentRef.current.contains(event.target)) {
+      toggle();
+    }
+  }
+
+  useEffect(() => {
+    const handleEscapeKeyPress = (event) => {
+      if (event.key === "Escape") {
+        toggle();
+      }
+    }
+
+    document.addEventListener("keydown", handleEscapeKeyPress);
+
+    return () => {

```

Comment:
Please can you check your VSCode is using our coding standards, I think you might need the Prettier plugin

Reply:
Sure, I'm making the changes